text/0000-scoped-slots-light-dom.md

```diff
@@ -0,0 +1,509 @@
+            <p>{slot2data.title}</p>
+        </template>
+
+        <template lwc:slot-data="defaultdata"> <!--// for default slot-->
+            <p>{defaultdata.title}</p>
+        </template>
+    </x-child>
+</template>
+
+// x/child.html
+<template lwc:render-mode="light">
+    <slot name="slotname1" lwc:slot-bind={slot1data}></slot>
+    <slot name="slotname2" lwc:slot-bind={slot2data}></slot>
+    <slot lwc:slot-bind={defaultdata}></slot>
+</template>
+```
+
+### Restricting Ambigious Bindings
+In the child component, the `slotname` to `lwc:slot-bind` binding is 1:1, but the opposite relationship is 1:n.
+* A component cannot bind its default scoped slot to different bindings.
+* A component cannot bind the same named slot to different bindings. 
+
+For example, these are invalid and will result in a compiler error:
+```html
+<template lwc:render-mode="light">
+    <slot name="slotname" lwc:slot-bind={slot1data}></slot>
+    <slot name="slotname" lwc:slot-bind={slot2data}></slot>
+</template>
+```
+```html
+<template lwc:render-mode="light">
+    <slot lwc:slot-bind={slot1data}></slot>
+    <slot lwc:slot-bind={slot2data}></slot>
+</template>
+``` 
+
+However, the same binding can be associated with multiple slots.
+```html
+<template lwc:render-mode="light">
+    <slot lwc:slot-bind={slotdata}></slot>
+    <slot name="slotname1" lwc:slot-bind={slotdata}></slot>
+    <slot name="slotname2" lwc:slot-bind={slotdata}></slot>
+</template>
+```
+_The last one is not ambigious, but for simplifying the restriction and conveying that "slot to binding combination should be unique", it will be restricted._
+
+### Mixing Standard Slots and Scoped Slots
+A parent component can set slot content into a child's standard slots and scoped slots. However, it cannot mix the two slot types for a given slot. In other words, the child cannot have two default slots, one for standard slot and another for scoped slot. Similarly, the child annot have two named slots of mixed slot type.
+#### Valid usages
+```html
+// x/parent.html
+<template>
+    <x-child>
+        <template slot="slotname1" lwc:slot-data="slot1data">
+            <p>{slot1data.name}</p>
+        </template>
+        <span>Chocolatier</span>
+    </x-child>
+</template>
+
+// x/child.html
+<template lwc:render-mode="light">
+    <slot name="slotname1" lwc:slot-bind={slot1data}></slot>
+    <slot></slot>
+</template>
+```
+
+#### Invalid Usages
+
+- Mixed default slots, one for regular slot and another for scoped slot
+```html
+// x/parent.html
+<template>
+    <x-child>
+        <template lwc:slot-data="slotdata">
+            <p>{slotdata.name}</p>
+        </template>
+        <p>Chocolatier</p>
+    </x-child>
+</template>
+
+// x/child.html
+<template lwc:render-mode="light">
+    <slot lwc:slot-bind={slotdata}>Scoped Slot</slot>
+    <slot>Regular Slot</slot>
+</template>
+```
+
+- Mixed named slots, one for regular slot and another for scoped slot

```

Comment:
Is it invalid because they share the same name? Or because the 2 of them are named slots?

Reply:
It's because they share the same name. The ambiguity here is:
1. In native shadow, if a child component has two slots with same name, all the slotted content appear at the first `<slot></slot>`. Should we follow this standard and enforce the same behavior on scoped slots?
2. Or should the engine be intelligent and match the slot content from the parent to the child's slot based on the slot type?
#2 definitely seems wrong IMO. #1 might be good. But to err on the side of caution, let us restrict it and may be open it up later if we find valid usecases.